### PR TITLE
Push latest images on make cluster-sync

### LINKED ISSUE #2380: cluster-sync only works the first time
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>

/kind bug

> /kind enhancement

**What happened**:

When we started to source `clustser-up` in #2335, the image synchronization in https://github.com/kubevirt/kubevirt/blob/release-0.17/cluster/ephemeral-provider-common.sh#L44 got lost. Now `cluster-sync` only pushes the images to the registry, but if images with this name are already on the nodes, they are not pulled down. As a result we still see the code from the initial sync running.

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- KubeVirt version (use `virtctl version`):
- Kubernetes version (use `kubectl version`):
- VM or VMI specifications:
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by cynepco3hahue**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2380

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: hack/config-default.sh [hunk 1]
```diff
@@ -7,7 +7,7 @@ cdi_namespace=cdi
 image_pull_policy=${IMAGE_PULL_POLICY:-IfNotPresent}
 verbosity=${VERBOSITY:-2}
 package_name=${PACKAGE_NAME:-kubevirt-dev}
-kubevirtci_git_hash="a4c50db1f73adeeea269861c082c43d10fa2e6fc"
+kubevirtci_git_hash="45cdf80da619866cdab36295de25a8aeea7eccbc"
```
Comment (davidvossel) [inline hack/config-default.sh]: did make generate run? i see the git hash changed but there's no code change in cluster-up.

File: hack/cluster-build.sh [hunk 2]
```diff
@@ -37,4 +37,30 @@ ${KUBEVIRT_PATH}hack/dockerized "DOCKER_PREFIX=${DOCKER_PREFIX} DOCKER_TAG=${DOC
...
+container=""
+container_alias=""
+for arg in ${docker_images}; do
+    name=$(basename $arg)
+    container="${container} ${manifest_docker_prefix}/${name}:${docker_tag} ${manifest_docker_prefix}/${name}:${docker_tag_alt}"
+    container_alias="${container_alias} ${manifest_docker_prefix}/${name}:${docker_tag} kubevirt/${name}:${docker_tag}"
+done
+
+# OKD provider has different node names and does not have docker
+if [[ $KUBEVIRT_PROVIDER =~ okd.* ]]; then
```
Comment (davidvossel) [inline hack/cluster-build.sh]: the crio ocp3 lane is podman as well.

File: hack/config-default.sh (hunk 1 above)
Comment (cynepco3hahue) [reply hack/config-default.sh]: done

File: hack/cluster-build.sh (hunk 2 above)
Comment (cynepco3hahue) [reply hack/cluster-build.sh]: os-3.11-crio still uses the docker to work with images so it should be fine

**Comment by rmohr** [comment]
> would it be worth us setting pullpolicy: "Always" just for our cluster-build manifests so we don't have to manually pull?

For as long as we don't use `Always` on virt-launcher, it would be ok. If virt-launcher is pulled with Always, it significantly increases the VMI start time.

**Pull request merged**